### PR TITLE
Recover coverage for relationship storage and core modules

### LINKED ISSUE #594: test(storage): recover coverage regression in relationship storage modules
### Summary
Recover and stabilize coverage in relationship storage and core relationship memory modules that regressed in commit `b160cbb`.

### Problem
The Codecov report shows notable regressions:
- `memory-storage-turso/src/relationships.rs` — 61.54% HEAD, 0.00% patch, -11.19%
- `memory-core/src/memory/relationships.rs` — 88.11% HEAD, 31.25% patch, -5.47%

These are the clearest coverage regressions in the commit.

### Why this matters
Relationship storage and core relationship memory logic can affect multiple command and retrieval surfaces. Regressions here have a wider blast radius than low coverage in isolated utility modules.

### Scope
Add focused tests for relationship storage and core memory relationship behavior.

### Proposed work
- Add tests for creating and retrieving relationships.
- Add tests for querying relationships by expected selectors or keys.
- Add tests for update or overwrite behavior if supported.
- Add tests for not-found and empty-result behavior.
- Add tests for invalid persistence or unsupported input paths.
- Add tests for conversion or mapping logic between core and storage layers if applicable.
- Add at least one regression test covering the changed lines or recently modified flow.

### Acceptance criteria
- `memory-storage-turso/src/relationships.rs` patch coverage improves above 0.00%.
- `memory-core/src/memory/relationships.rs` patch coverage improves above 31.25%.
- The coverage drop in both modules is reduced in the next Codecov run.
- Tests cover success, empty-result, and failure scenarios.

### Notes
Use the narrowest possible test harness. Add integration-style setup only where module-level tests cannot exercise the important behavior.

### PR DESCRIPTION
**Pull request by d-o-hub**
This change restores test coverage for the relationship management modules in both `do-memory-core` and `do-memory-storage-turso`. It introduces new tests for CRUD operations, global retrieval, cycle detection, and graph building, ensuring that recent regressions are addressed and the logic is stabilized.

Fixes #594

---
*PR created automatically by Jules for task [15046510184985316669](https://jules.google.com/task/15046510184985316669) started by @d-o-hub*

### CODE REVIEW

File: memory-core/src/memory/relationships/tests.rs [hunk 1]
```diff
@@ -122,3 +122,265 @@ async fn test_get_episode_with_relationships() {
...
+            TaskType::Testing,
+        )
+        .await;
+    let ep2 = memory
+        .start_episode(
+            "Task 2".to_string(),
+            TaskContext::default(),
+            TaskType::Testing,
+        )
+        .await;
```
[the remark this answers is not in the document]
Comment (d-o-hub) [reply memory-core/src/memory/relationships/tests.rs]: @jules
